src/toxcore/dht/packet.rs

```diff
@@ -1602,6 +1447,16 @@ mod tests {
                 };
                 let decoded_payload = invalid_packet.get_payload(&bob_sk);
                 assert!(decoded_payload.is_err());
+                // Try short incomplete for *Responses
+                let invalid_payload = [0x01];
```

Comment:
Why do you need to distinguish requests and responses? Maybe empty payload will fit for both?

Reply:
Anyway this part won't be necessary after migration to nom 4.

Reply:
I write that code to fix this problem which is mentioned by Roman 
"This one is missing too https://coveralls.io/builds/15940383/source?filename=toxcore%2Fdht%2Fpacket.rs#L340"
If there is any other good solution, let me know.

Reply:
Did you try to use invalid_payload = []; ? Because it seems to me that it's not quite correct to use 0x00 and 0x01 for requests/responses because it covers only ping/pong cases.

Reply:
OK!
I will try your recommendation on next PR